lib/controller.dart

```diff
@@ -0,0 +1,68 @@
+library controller;
+
+import 'package:dashboard/model.dart' as model;
+import 'package:dashboard/view.dart' as view;
+import 'package:dashboard/firebase.dart' as fb;
+import 'package:dashboard/logger.dart';
+
+Logger logger = Logger('controller.dart');
+
+var _navLinks = [
+  model.Link('analyse', 'Analyse', view.renderAnalyseTab),
+  model.Link('settings', 'Settings', view.renderSettingsTab)
+];
+var _currentNavLink = _navLinks.first.pathname;
+
+// Actions
+enum UIAction { signinWithGoogle, changeNavTab }
+
+// Action data
+class Data {}
+
+class NavChangeData extends Data {
+  String pathname;
+  NavChangeData(this.pathname);
+}
+
+// Controller functions
+void init() async {
+  view.init();
+  view.showLoginModal();
+  for (var n in _navLinks) {
+    view.appendNavLink(n.pathname, n.label, _currentNavLink == n.pathname);
+  }
+
+  await fb.init(
+      'assets/firebase-constants.json', onLoginCompleted, onLogoutCompleted);
+}
+
+void onLoginCompleted() async {
+  await loadDataFromDB();
+  view.setNavlinkSelected(_currentNavLink);
+  _navLinks.first.render();
+}
+
+void loadDataFromDB() async {
```

Comment:
nit: even though we use Firebase as a database, we've consistently referred to it by name rather than purpose - consider renaming this method to `loadDataFromFirebase`

Reply:
Got it, fixed